mlir/lib/Dialect/Rock/Transforms/ThreadwiseGemmLowering.cpp

```diff
@@ -547,6 +547,7 @@ LogicalResult ThreadwiseReadIntoRewritePattern::matchAndRewrite(
   if (isSrcVectorBuffer) {
     loadType = elementType;
     vectorSrcLen = elementType.dyn_cast<VectorType>().getNumElements();
+    elementType = elementType.dyn_cast<VectorType>().getElementType();
```

Comment:
Refactoring nit: instead of these `isa`/`dyn_cast` pairs (which should be `cast` here anyway), we could instead go `VectorType maybeSrcBufferVectorType = dyn_cast<VectorType>(...)` above